### DISCUSSION TITLE
CSV download endpoint returning 404 v2

### DISCUSSION THREAD

**Discussion by KevinMind** [socketlabs/discussions#8] [Q&A] [answered]
Hey folks. I'm attempting to use the officially documented socketlabs api spec to download a suppression list to a csv file.

I'm following this spec

and running this code, (replace the API key and server id)

```javascript
#!/usr/bin/env node
/* eslint-disable global-require, no-console */

import fs from 'fs';

import { Headers } from 'node-fetch';

const api_key = '<token>";
const server_id = '<serverId>';

const myHeaders = new Headers();
myHeaders.append('Authorization', `Bearer ${api_key}`);

const requestOptions = {
  method: 'GET',
  headers: myHeaders,
  redirect: 'follow',
};

fetch(`https://api.socketlabs.com/v2/server/${server_id}/suppressions/download?sortField=suppressionLastUpdate&sortDirection=dsc`, requestOptions)
  .then((response) => response.text())
  .then((result) => console.log(result))
  .catch((error) => console.log('error', error));
```

I execute this way

```bash
node ./bin/socket-labs.mjs
```

using node18

I get this response

```bash
{"error":[{"errorType":"NotFound","message":"The resource you were requesting could not be found."}]}
```

I've tried replacing the URL for the json endpoint of the same resource. To be prceise this endpoint:

`https://api.socketlabs.com/v2/servers/:serverId/suppressions/search?pageSize=5&pageNumber=0&sortField=emailAddress&sortDirection=dsc`

And it returned a valid json response.

I tried changing the url to an invalid one:

```javascript
https://api.socketlabs.com/v2/server/${server_id}/banana/download?sortField=suppressionLastUpdate&sortDirection=dsc
```

and get the exact same 404 So I feel like there is a routing bug on socketlabs side... is this possible?

**Comment by KevinMind** [comment]
I have tried executing via curl and python, same result.

**Comment by KevinMind** [comment, marked as answer]
I found the issue... There are typos in your documentation https://docs.socketlabs.com/

Look carefully:
```tsx
server/${server_id}/b
```

"server" should actually be "servers"

Can you update this?

**Reply by david-schrenker** [reply]
thank you for finding the error, the documentation has been updated.
